### PR TITLE
[TEP-0075] Add variable usage and links to examples in docs

### PR DESCRIPTION
**Pull request by chuangw6**
# Changes

- Added instructions on how object variables can be used on Pipeline
and Task level.
- Added links to example yaml files in `docs/tasks.md`.

/kind misc

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [ ] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Update docs
```

### CODE REVIEW

File: docs/tasks.md [hunk 1]
```diff
@@ -555,13 +553,6 @@ metadata:
   name: task-with-parameters
 spec:
   params:
-    - name: gitrepo
```
Comment (lbernick) [inline docs/tasks.md:558]: Curious what's the reason to remove this? I think it's useful to have example usage in the docs as well as in the examples folder

Comment (chuangw6) [reply docs/tasks.md]: sg! added them back. Thanks

File: docs/tasks.md [hunk 2]
```diff
@@ -527,14 +527,12 @@ For example, `foo.Is-Bar_` is a valid parameter name for string or array type, b
...
-
-
+  > - When using object in variable replacement, users can only access its individual key ("child" member) of the object by its name i.e. `$(params.gitrepo.url)`. Using an entire object as a value is only allowed when the value is also an object like [this example](https://github.com/tektoncd/pipeline/blob/55665765e4de35b3a4fb541549ae8cdef0996641/examples/v1beta1/pipelineruns/alpha/pipeline-object-param-and-result.yaml#L64-L65). See more details about using object param from the [TEP-0075](https://github.com/tektoncd/community/blob/main/teps/0075-object-param-and-result-types.md#using-objects-in-variable-replacement).
 
 - `array` type 
 `array` type is useful in cases where the number of compilation flags being supplied to a task varies throughout the `Task's` execution.
```
Comment (lbernick) [inline docs/tasks.md:538]: this paragraph needs some updates for object params

Comment (chuangw6) [reply docs/tasks.md:538]: Done.

**Review by lbernick — APPROVED** [review]

**Pull request merged**